### ISSUE TITLE
else-if indent broken after fix

### ISSUE THREAD

**Issue by djibarian** [eslint/eslint#17642]
### What did you expect to happen?

Indent here is OK, rule shouldn't complain.

```
if(foo)
    doSomething(true);
else
if(bar)
   doSomething(false);
```

### What actually happened?

As explained in https://github.com/eslint/eslint/issues/17316#issuecomment-1639407430, the rule gives a false positive with the following code:

```
if(foo)
    doSomething(true);
else
if(bar)
   doSomething(false);
```

saying that the indentation at `if(bar)` is wrong.
